tracer/src/Datadog.Trace/AppSec/Waf/Context.cs

```diff
@@ -75,21 +81,49 @@ private Context(IntPtr contextHandle, Waf waf, WafLibraryInvoker wafLibraryInvok
 
             if (Log.IsEnabled(LogEventLevel.Debug))
             {
-                var parameters = Encoder.FormatArgs(addresses);
-                Log.Debug("DDAS-0010-00: Executing AppSec In-App WAF with parameters: {Parameters}", parameters);
+                var persistentParameters = persistentAddressData == null ? string.Empty : Encoder.FormatArgs(persistentAddressData);
+                var ephemeralParameters = ephemeralAddressData == null ? string.Empty : Encoder.FormatArgs(ephemeralAddressData);
+                Log.Debug(
+                    "DDAS-0010-00: Executing AppSec In-App WAF with persistent parameters: {PersistentParameters}, ephemeral parameters: {EphemeralParameters}",
+                    persistentParameters,
+                    ephemeralParameters);
             }
 
             // not restart cause it's the total runtime over runs, and we run several * during request
             _stopwatch.Start();
             WafReturnCode code;
             lock (_stopwatch)
             {
-                var args = _encoder.Encode(addresses, applySafetyLimits: true);
-                var pwArgs = args.Result;
-                _argCache.Add(args);
-
-                // WARNING: DO NOT DISPOSE pwArgs until the end of this class's lifecycle, i.e in the dispose. Otherwise waf might crash with fatal exception.
-                code = _waf.Run(_contextHandle, pwArgs, ref retNative, timeoutMicroSeconds);
+                IEncodeResult? persistentArgs = null;
+                IntPtr pwPersistentArgs = IntPtr.Zero;
+                if (persistentAddressData != null)
```

Comment:
```suggestion
               if (persistentAddressData is { Count: > 0 })
```

Reply:
Allowing an empty dictionary will ensure the WAF runs without returning an error. We shouldn't really call this method with an empty dictionary, but we have some empty unit tests.

Reply:
same as other comment on returning null but instead but Im not opinionated about it :)

Reply:
I think return null when both persistent and ephemeral are absent is enough.